ionoscloud/resource_server.go

```diff
@@ -592,6 +592,15 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
 			diags := diag.FromErr(fmt.Errorf("error while setting primary nic %s: %w", d.Id(), err))
 			return diags
 		}
+		if (*server.Entities.Nics.Items)[0].Properties != nil &&
```

Comment:
if you want to avoid the chances of bugs occurring during a refactor or when you write the code, you can store the code sequence: *server.Entities.Nics.Items)[0].Properties.Ips  in a variable(for example: serverIps).
Plus this is a good practice because you avoid spaghetti code too so the code is easier to read.